app/controllers/action/AuthorisedSaUserAction.scala

```diff
@@ -18,51 +18,66 @@ package controllers.action
-  extends WrappedRequest[A](request)
+final class AuthorisedSaUserRequest[A](val request: AuthenticatedRequestWithJourney[A], val utr: SaUtr)
+  extends WrappedRequest[A](request) {
+
+  val journey: Journey = request.journey
+}
 
 class AuthorisedSaUserAction @Inject() (
     appConfig: AppConfig,
     cc:        MessagesControllerComponents)
-  extends ActionRefiner[AuthenticatedRequest, AuthorisedSaUserRequest] {
-
-  override protected def refine[A](request: AuthenticatedRequest[A]): Future[Either[Result, AuthorisedSaUserRequest[A]]] = {
+  extends ActionRefiner[AuthenticatedRequestWithJourney, AuthorisedSaUserRequest] {
 
+  override protected def refine[A](request: AuthenticatedRequestWithJourney[A]): Future[Either[Result, AuthorisedSaUserRequest[A]]] = {
+    implicit val r: Request[_] = request
+    implicit val ec: ExecutionContext = executionContext
     val hasActiveSaEnrolment: Boolean = request.hasActiveSaEnrolment
     val maybeUtr: Option[SaUtr] = request.maybeUtr
     val isConfident: Boolean = request.confidenceLevel >= ConfidenceLevel.L200
+    import req.RequestSupport._
+
+    //this is being updated by notification from add-taxes-frontend
+    //at some point if user was sent to add-taxes-frontend for enrol-for-sa journey
+    val enrolledForSa = request.journey.userEnrolledForSa.getOrElse(false)
 
-      def logFail(reason: String) = Logger.info(
+      def logFail(reason: String) = JourneyLogger.info(
         s"""
-         |Authorisation failed, reason: $reason:
+         |Authorisation failed. $reason.
          |  [hasActiveSaEnrolment: $hasActiveSaEnrolment]
          |  [enrolments: ${request.enrolments}]
          |  [utr: $maybeUtr]
          |  [ConfidenceLevel: ${request.confidenceLevel}]
-         |  """.stripMargin
+         |  """.stripMargin,
+        request.journey
       )
 
     val result: Either[Result, AuthorisedSaUserRequest[A]] =
       (hasActiveSaEnrolment, maybeUtr, isConfident) match {
         case (false, _, _) =>
-          logFail("no active IR-SA enrolment")
+          logFail("no active IR-SA enrolment, sending user to add-taxes-frontend for enrol-for-sa journey")
           Left(Redirect(ssttpeligibility.routes.SelfServiceTimeToPayController.getAccessYouSelfAssessmentOnline()))
         case (_, None, _) =>
-          logFail("no present UTR")
+          logFail("no present UTR, sending user to add-taxes-frontend for enrol-for-sa journey")
           Left(Redirect(ssttpeligibility.routes.SelfServiceTimeToPayController.getAccessYouSelfAssessmentOnline()))
         case (true, Some(utr), false) =>
-          logFail("Confidence level is to low, redirecting to identity-verification")
-          Left(redirectToUplift(request))
+          if (enrolledForSa) {
+            JourneyLogger.info("Confidence level is to low, but user came back from add-taxes-frontend from enrol-for-sa journey thus allowing him to proceed.", request.journey)
```

Comment:
pedantic: typo in "too", same in else block

Reply:
Just noting a vague question coming to mind, but I expect the answer will be clearer after looking at the rest of the PR...

Q: are there any other confidence level reference/checks further on in the journey/codebase, or are they all captured in this action? If there are other places that check it, they'd also likely need to check this `enrolledForSa` value too